### PR TITLE
:bug: Fix the response structure: TEXT -> JSON

### PR DESCRIPTION
**Pull request by huyfififi**
## I'm submitting a ...

- bug fix

## What is the current behavior?

plain text is returned (?)

## What is the expected behavior?

JSON response is returned

## Checklist

- [x] tests are present (if necessary)
- [x] no commented out code
- [x] no debbugging code
- [x] errors are handled
- [x] properly typed

### CODE REVIEW

File: backend/src/backend/app.py [hunk 1]
```diff
@@ -157,11 +162,9 @@ async def chat(chat_request: ChatRequest):
 
     response = openai.chat.completions.create(
         model="gpt-3.5-turbo",
-        messages=[
```
Comment (huyfififi) [inline backend/src/backend/app.py:160]: format changes made by `black`

**Comment by huyfififi** [comment]
@LucileTech do you have bandwidth to review this?

File: backend/src/backend/tests.py [hunk 2]
```diff
@@ -45,28 +45,52 @@ def test_list_tasks_unauthorized_user():
 def test_chat_endpoint_malformed_request():
     response = client.post("/api/chat", json={"test": "test"})
     assert response.status_code == 400
-    assert "type" in response.text
-    assert "loc" in response.text
-    assert "msg" in response.text
+    assert response.headers["Content-Type"] == "application/json"
```
Comment (huyfififi) [inline backend/src/backend/tests.py:48]: It was `text/plain`, but JSON response is the industrial standard

**Comment by LucileTech** [comment]
> @LucileTech do you have bandwidth to review this?
JSON response is returned as wanted

Screenshot 2024-08-27 at 09 50 31

Also, the front end still has the expected behavior
Screenshot 2024-08-27 at 09 51 03

**Review by LucileTech — APPROVED** [review]

**Pull request merged**